### ISSUE TITLE
[Feature Request] Support for willImplement

### ISSUE THREAD

**Issue by alexander-schranz** [Jan0707/phpstan-prophecy#66]
I have the following code:

```
$contentView = $this->prophesize(ExcerptInterface::class);
$contentView->willImplement(ContentViewInterface::class);
$contentView->getExcerptTitle()->willReturn('Excerpt Title');

$merger->merge($contentView->reveal(), ..);
```

This will error with:

> `Parameter #1 $contentView of method MergerInterface::merge() expects ContentViewInterface, ExcerptInterface given`

Not sure if this is easy possible to be integrated. Or do you have any hint, I sadly have no knowledge about phpstan.

**Comment by Jan0707** [comment]
From what I understand you could (should) simply do this:

```
$contentView = $this->prophesize(ContentViewInterface::class);
$contentView->getExcerptTitle()->willReturn('Excerpt Title');

$merger->merge($contentView->reveal(), ..);
```

As this is clearly the interface whose behaviour your testing for here.

Would that be a feasible solution for you ?

**Comment by alexander-schranz** [comment]
@Jan0707 then it fails on getExcerptTitle as that is implemented by ExcerptInterface only so swithcing prophesize and willImplement does just move the error to another line.

**Comment by localheinz** [comment]
@alexander-schranz 

Just released [`jangregor/phpstan-prophecy:0.6.0`](https://github.com/Jan0707/phpstan-prophecy/releases/tag/0.6.0) with support for 

- `willExtend()` 
- `willImplement()`

Let me know if this works for you or if you encounter any issues!

🤓
